tools/test/donkeyrider.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Copyright European Organization for Nuclear Research (CERN) since 2012
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+import shutil
+import signal
+import sys
+
+stop = False
+
+
+def shutdown(signum, frame):
+    global stop
+    stop = True
+    print("Caught", signal.Signals(signum).name + ". Stopping..", file=sys.stderr)
+    sys.exit(1)
+
+
+def main():
+    if 'USE_PODMAN' not in os.environ or os.environ['USE_PODMAN'] == "":
+        docker_executable = shutil.which("docker")
+        podman_executable = shutil.which("podman")
+        if docker_executable and podman_executable:
+            with open(docker_executable, 'br') as fhandle:
+                if podman_executable.encode() in fhandle.read():
+                    os.environ['USE_PODMAN'] = "1"
+        elif docker_executable:
+            os.environ['USE_PODMAN'] = "0"
+        elif podman_executable:
+            os.environ['USE_PODMAN'] = "1"
+        else:
+            print("No compatible container executable (podman/docker) found! Exiting..", file=sys.stderr)
+            sys.exit(1)
+
+    project_dir = os.path.abspath(os.path.join(os.path.abspath(os.path.dirname(__file__)), "../.."))
+    print("Detected project directory", project_dir, file=sys.stderr)
```

Comment:
Writing to stderr as debug does not make a lot of sense from my POV. How about using `logging`?

Reply:
Using stderr as logging is very common in posix environments. I don't think it is necessary to use logging for a few lines of logging inbetween. This is a developer tool after all.